src/context/config.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  createContext,
+  useContext,
+  FunctionComponent,
+  useState,
+  useEffect
+} from "react";
+import { AsyncStorage } from "react-native";
+import { LoadingView } from "../components/Loading";
+
+export enum AppMode {
+  production = "production",
+  staging = "staging"
+}
+
+interface Config {
+  appMode: AppMode;
+}
+
+interface ConfigContext {
+  config: Config; // context consumers should never get an undefined config
+  setConfigValue: <K extends keyof Config>(key: K, value: Config[K]) => void;
+}
+
+const CONFIG_KEY = "CONFIG";
+const DEFAULT_CONFIG: Config = { appMode: AppMode.production };
+
+const ConfigContext = createContext<ConfigContext>({
+  config: DEFAULT_CONFIG,
+  setConfigValue: () => {} // eslint-disable-line @typescript-eslint/no-empty-function
+});
+
+export const useConfigContext = (): ConfigContext =>
```

Comment:
you could have kept useConfig as a name

Reply:
I left the `Context` in there to remind the consumer not use this without ensuring an enclosed context.